api/room/schema_query.py

```diff
@@ -256,6 +267,71 @@ class Query(graphene.ObjectType):
                 [required]"
     )
 
+    all_available_rooms = graphene.Field(
+        AvailableRooms,
+        start_date=graphene.String(required=True),
+        start_time=graphene.String(required=True),
+        end_date=graphene.String(required=True),
+        end_time=graphene.String(required=True),
+        time_zone=graphene.String(required=True),
+        location=graphene.String(required=True),
+        description="Returns available rooms for a specific time \
+            \n- start_time: Start time and date when you want to book room from\
+            [required]\n- end_time: time and date you want to book room upto\
+                [required]\n time_zone: The time zone of the location\
+                    [required]\n location: The location of the office's \
+                        room you want to book"
+    )
+
+    def resolve_all_available_rooms(self, info, **kwargs):
+
+        validate_date_input(kwargs['start_date'], kwargs['end_date'])
+        validate_date_input(kwargs['start_time'], kwargs['end_time'])
+
+        format_range_dates(kwargs['start_date'], kwargs['end_date'])
+        format_range_time(kwargs['start_time'], kwargs['end_time'])
+
+        new_start_time = convert_date(kwargs['start_date'],
+                                      kwargs['start_time'],
+                                      kwargs['time_zone']
+                                      )
+        new_end_time = convert_date(kwargs['end_date'],
+                                    kwargs['end_time'],
+                                    kwargs['time_zone']
+                                    )
+
+        booked_rooms = EventsModel.query.with_entities(
+            EventsModel.room_id,
+            ).filter(
+            EventsModel.state == "active",
+            EventsModel.start_time >= new_start_time,
+            EventsModel.end_time <= new_end_time
+        ).all()
+
+        rooms_location = LocationModel.query.with_entities(
+            LocationModel.id).filter(
+            LocationModel.name == kwargs['location'],
```

Comment:
I think it will be easier if you use location id instead of location name and you should also check if that location exists maybe.